pkg/tbtc/registry.go

```diff
@@ -156,6 +156,48 @@ func (wr *walletRegistry) getWalletByPublicKeyHash(
 	return wallet{}, false
 }
 
+// archiveWallet archives the wallet with the given public key hash. The wallet
+// data is removed from the wallet cache and the entire wallet storage directory
+// is moved to the archive directory.
+func (wr *walletRegistry) archiveWallet(
+	walletPublicKeyHash [20]byte,
+) error {
+	wr.mutex.Lock()
+	defer wr.mutex.Unlock()
+
+	var walletPublicKey *ecdsa.PublicKey
+
+	for _, value := range wr.walletCache {
+		if value.walletPublicKeyHash == walletPublicKeyHash {
+			// All signers belong to one wallet. Take the wallet public key from
+			//  the first signer.
+			walletPublicKey = value.signers[0].wallet.publicKey
```

Comment:
Shouldn't we `break` the loop once `walletPublicKey` is set?

Reply:
Good catch.
Done in ce85a89584bdb6a6a337f65ead9f32c999d220a0.